.github/workflows/main.yml

```diff
@@ -9,10 +9,47 @@ on:
 
 jobs:
   build:
-    uses: ./.github/workflows/ci.yml
+    runs-on: windows-latest
+    outputs:
+      version: ${{ steps.set-version.outputs.version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.2xx # Align with global.json (including roll forward rules)
+      
+      - name: Cache Nuget     
+        uses: actions/cache@v4
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-nuget-
+
+      - name: ⚒️ Run GitVersion
+        run: ./build.ps1 build-server-version
+
+      - name: ⚒️ Run build
+        run: ./build.ps1
+
+      - name: ⬆️ Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: output-${{ env.GitVersion_FullSemVer }}
+          path: output/*.*
+          compression-level: 0 # no compression
 
+      - id: set-version
+        name: Set version to output
+        run: echo "version=${{ env.GitVersion_FullSemVer }}" >> "$GITHUB_OUTPUT"
```

Comment:
Why the copy/paste? reusable workflows are ♥️
If we need some variation we can add inputs to it.